### PR TITLE
adds all federal states for germany

### PR DESCRIPTION
**Pull request by PhilippSchreiber**

### CODE REVIEW

**Comment by PhilippSchreiber** [comment]
I have no idea about the copyright... So please change it if you want to mention my little work or leave it as is...

**Comment by stedekay** [comment]
Is it possible to merge this PR?
Could need these additions.

File: src/Holiday/Brandenburg.php [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+namespace Holiday;
+
+class Brandenburg extends Germany
+{
+    protected function getHolidays($year)
+    {
+        $timezone = $this->timezone;
+
+        $data   = parent::getHolidays($year);
+        $data[] = new Holiday("6.1." . $year, "Dreikönigstag", $timezone);
```
Comment (calbrecht) [inline src/Holiday/Brandenburg.php]: According to http://feiertage.net/bundeslaender.php Brandenburg has no "Dreikönigstag", even https://de.wikipedia.org/wiki/Gesetzliche_Feiertage_in_Deutschland#.C3.9Cbersicht_aller_gesetzlichen_Feiertage agrees with that. What is the source of your implementation?

File: src/Holiday/BadenWuerttemberg.php [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+class BadenWuerttemberg extends Germany
+{
+    protected function getHolidays($year)
+    {
+        $timezone = $this->timezone;
+
+        $easter = $this->getEaster($year);
+        $data   = parent::getHolidays($year);
+        $data[] = new Holiday("6.1." . $year, "Dreikönigstag", $timezone);
```
Comment (calbrecht) [inline src/Holiday/BadenWuerttemberg.php]: The legislative term according to wikipedia https://de.wikipedia.org/wiki/Gesetzliche_Feiertage_in_Deutschland#.C3.9Cbersicht_aller_gesetzlichen_Feiertage is "Heilige Drei Könige"

File: src/Holiday/Saxony.php [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+{
+    protected function getHolidays($year)
+    {
+        $timezone = $this->timezone;
+
+        $data   = parent::getHolidays($year);
+
+        $data[] = new Holiday("31.10." . $year, "Reformationstag", $timezone);
+
+        $date = new Holiday("24.12." . $year, "Buß- und Bettag", $timezone);
```
Comment (calbrecht) [inline src/Holiday/Saxony.php]: I think you may simplify this calculation with "23.11" and modify "last Wednesday" according to https://de.wikipedia.org/wiki/Gesetzliche_Feiertage_in_Deutschland#.C3.9Cbersicht_aller_gesetzlichen_Feiertage and https://de.wikipedia.org/wiki/Bu%C3%9F-_und_Bettag

**Comment by calbrecht** [comment]
Will be happy to merge once the additional Holiday in Brandenburg is gone. 
If you are willing please change "Dreikönigstag" to "Heilige Drei Könige" as it seems to be the common term for that Holiday. Also simplifying thew calculation on "Buß- und Bettag" would be nice to have.
Thank you.

File: src/Holiday/BadenWuerttemberg.php (hunk 2 above)
Comment (PhilippSchreiber) [inline src/Holiday/BadenWuerttemberg.php]: All people I know say "Heilige Drei Könige", never heard about "Erscheinungsfest" and I don't find the source for the term in BW

File: src/Holiday/Brandenburg.php (hunk 1 above)
Comment (PhilippSchreiber) [inline src/Holiday/Brandenburg.php]: Fixed

File: src/Holiday/Saxony.php (hunk 3 above)
Comment (PhilippSchreiber) [inline src/Holiday/Saxony.php]: Done. Tests are still good. :)

**Comment by PhilippSchreiber** [comment]
I added the special "Reformationstag 2017", too. https://de.wikipedia.org/wiki/Gesetzliche_Feiertage_in_Deutschland#cite_note-reform2-13

**Pull request merged**